target/linux/ramips/dts/mt7620a_humax_e2.dts

```diff
@@ -0,0 +1,144 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "mt7620a.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/mtd/partitions/uimage.h>
+
+/ {
+	compatible = "humax,e2", "ralink,mt7620a-soc";
+	model = "HUMAX E2";
+
+	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+		label-mac-device = &ethernet;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power: power {
+			label = "green:power";
+			gpios = <&gpio1 3 GPIO_ACTIVE_LOW>;
+		};
+
+		pairing {
+			label = "green:pairing";
+			gpios = <&gpio1 8 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan {
+			label = "green:wlan";
+			gpios = <&gpio1 9 GPIO_ACTIVE_LOW>;
+		};
+
+		lan {
+			label = "green:lan";
+			gpios = <&gpio1 4 GPIO_ACTIVE_LOW>;
+		};
+
+		wps {
+			label = "green:wps";
+			gpios = <&gpio1 5 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio1 1 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+};
+
+&gpio1 {
+	status = "okay";
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
```

Comment:
Can it go faster? 50MHz might be a good candidate.

Reply:
It's just 'Copy and Paste' code. Corrected!

Reply:
No, it's not a correction, it's an improvement. You should test if the higher frequency does work with your device and RW speed has really improved (by dd'ing mtd). (BTW your freq is now 5MHz ;) )

Reply:
OMG You're right! I'll set to 50MHz!
Anyway, I have no time to test Higher Frequency, so I just match that to datasheet!
(In my E10 experience -It also have same manufacturer's SPI flash-, 100MHz also works good but well, I don't know.)

Reply:
> Anyway, I have no time to test Higher Frequency, so I just match that to datasheet!

No, test is a MUST, though it seems quite trivial. Malfunctioning flash can lead to soft-brick, and it cannot be assured just by looking at datasheet. (refer to #3159)
Take your time, I usually give a few weeks for new device support requests to get enough feedback.

Reply:
> > Anyway, I have no time to test Higher Frequency, so I just match that to datasheet!
> 
> No, on-board test is a MUST, though it seems quite trivial. Malfunctioning flash can lead to soft-brick, and it cannot be assured just by looking at datasheet. (refer to #3159) Take your time, I usually give a few weeks for new device support requests to get enough feedback.

I mean, Higher Frequency = over 50MHz.
Yes, I will test this next week. but I don't have more time to verify faster (fastest?) frequency.

Reply:
I tested now! It works perfectly!